server/src/infra/repositories/person.repository.ts

```diff
@@ -50,7 +50,15 @@ export class PersonRepository implements IPersonRepository {
     return people.length;
   }
 
-  getAll(userId: string, options?: PersonSearchOptions): Promise<PersonEntity[]> {
+  getAll(): Promise<PersonEntity[]> {
+    return this.personRepository.find();
+  }
+
+  getAllWithoutThumbnail(): Promise<PersonEntity[]> {
+    return this.personRepository.createQueryBuilder('person').where('LENGTH(person.thumbnailPath) = 0').getMany();
```

Comment:
Other places in the code, such as [asset.repository.ts](https://github.com/immich-app/immich/blob/main/server/src/infra/repositories/asset.repository.ts#L173-L180), use `IsNull()` for the WHERE rather than `LENGTH`.

Reply:
You can do = '' instead then